apps/prairielearn/src/pages/home/home.tsx

```diff
@@ -121,15 +127,42 @@ router.post(
     } = getPageContext(res.locals, { withAuthzData: false });
 
     if (body.__action === 'accept_invitation') {
-      await execute(sql.accept_invitation, {
+      await ensureEnrollment({
         course_instance_id: body.course_instance_id,
-        uid,
         user_id,
+        agent_user_id: user_id,
+        agent_authn_user_id: user_id,
+        action_detail: 'invitation_accepted',
       });
     } else if (body.__action === 'reject_invitation') {
-      await execute(sql.reject_invitation, {
+      const oldEnrollment = await selectOptionalCourseInstanceEnrollmentByPendingUid({
         course_instance_id: body.course_instance_id,
-        uid,
+        pending_uid: uid,
+      });
+
+      if (!oldEnrollment) {
+        throw new Error('Could not find enrollment to reject');
```

Comment:
This should use an `HttpStatusError` with an appropriate status code. Clients aren't to be trusted, it shouldn't be a 500 if someone throws a random request at us.

Reply:
I added a sentence there saying "We also want to improve the error handling for trying to invite/reject random ids." and linked back to this thread.